### PR TITLE
Fill the windows in the examples with a solid color

### PR DESCRIPTION
**Pull request by notgull**
- [ ] Tested on all platforms changed (only tested on X11)
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

When you run the examples for `winit`, the contents of the window will be undefined depending on the platform. To the uninformed user, this appears to be a bug in some cases. In the past we've gotten quite a few issues because of this (#2807 comes to mind) that aren't issues at all.

This PR fills the windows with a solid gray background in order to make sure that there is *something* inside of the window.

Before (X11):

image

After (X11):

image

### CODE REVIEW

**Comment by kchibisov** [comment]
That's true, though there's always glutin with window example, which I use as a testing or alacritty branches which I always open to target latest winit dev, since that's what I use myself as a maintainer to test winit API.

**Comment by notgull** [comment]
I've rebased on the latest master. So far `softbuffer` doesn't support Android (rust-windowing/softbuffer#44) or iOS (rust-windowing/softbuffer#43) yet, so I've just made `fill_window` a no-op on those platforms. The Windows CI errors are due to `winit` using `windows-sys` version 0.45 and should be fixed by #2842

**Review by kchibisov — CHANGES_REQUESTED** [review]
All examples must handle `Resize` events, otherwise Wayland won't fly.

Probably macOS won't fly as well.

File: examples/util/fill.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    GC.with(|gc| {
+        // Either get the last context used or create a new one.
+        let mut gc = gc.borrow_mut();
+        let context = gc.entry(window.id()).or_insert_with(|| unsafe {
+            GraphicsContext::new(window, window)
+                .expect("Failed to create a softbuffer graphics context")
+        });
+
+        // Fill a buffer with a solid color.
+        const LIGHT_GRAY: u32 = 0xFFD3D3D3;
```
Comment (kchibisov) [inline examples/util/fill.rs:33]:
```suggestion
        const LIGHT_GRAY: u32 = 0xFF181818;
```

my eyes.

**Comment by kchibisov** [comment]
>All examples must handle Resize events, otherwise Wayland won't fly.

>Probably macOS won't fly as well.


Actually, you do `Resize`, but the thing is that it's not really resized on Wayland, even when winit is telling the right sizes. I'd assume we'd need a new release of softbuffer?

cc @ids1024

**Comment by ids1024** [comment]
At this point we might as well do another release of softbuffer, from the current master, instead of having things relying on 0.2. It's a breaking release, and it will probably have another breaking release for other things, but that's not really an issue at this stage.

**Comment by kchibisov** [comment]
>At this point we might as well do another release of softbuffer, from the current master, instead of having things relying on 0.2. It's a breaking release, and it will probably have another breaking release for other things, but that's not really an issue at this stage.

That would be nice, given that it's un-usable in the current state on Wayland from what I've seen in this PR. The example on softbuffer master does work fine.

**Comment by notgull** [comment]
Now `cargo-deny` is flagging the `memmap2` dependency. Shouldn't be a real issue.

**Comment by kchibisov** [comment]
Could you add memmap2 into deny and open an issue on client-toolkit(I can do this myself if you don't want to)?

**Review by kchibisov — APPROVED** [review]

**Pull request merged**